csvcubed/csvcubed/cli/build.py

```diff
@@ -0,0 +1,79 @@
+"""
+Build Command
+-------------
+Build a qb-flavoured CSV-W from a config.json and a tidy CSV.
+"""
+import dataclasses
+import json
+import logging
+from pathlib import Path
+from typing import Optional
+
+from csvcubedmodels.dataclassbase import DataClassBase
+
+from csvcubed.readers.configdeserialiser import (
+    get_cube_from_data,
+    get_cube_from_config_json
+)
+from csvcubed.utils.qb.validation.cube import validate_qb_component_constraints
+
+_logger = logging.getLogger(__name__)
+_logger.addHandler(logging.StreamHandler())
+
+def build(
+    config: Path,
+    output_directory: Path,
+    csv_path: Path,
+    fail_when_validation_error_occurs: bool,
+    validation_errors_file_out: Optional[Path],
+):
+
+    # Create a path to save the validation-errors.json file in the ./out directory
+
+    _logger.debug(f"CSV: {csv_path.absolute() if csv_path is not None else ''}")
+    _logger.debug(f"qube-config.json: {config.absolute() if config is not None else ''}")
+
+    if config:
+        cube, json_schema_validation_errors = get_cube_from_config_json(config, csv_path)
```

Comment:
I think there's some misunderstanding here. We don't want an either/or approach to convention vs configuration - we want users to be able to mix and match convention and configuration. You should be able to rely on some columns using the conventional approach and other columns in the same cube to be configured in the config.json.

Reply:
refactored, abstracting some checks and combining the column processing to support mixed config / convention column creation.